packages/back-end/src/entities/database.ts

```diff
@@ -0,0 +1,234 @@
+        { id: id }
+      );
+
+      const father = result.records[0].get("f");
+
+      return father;
+    } finally {
+      await session.close();
+    }
+  }
+
+  async findMother(person: number | Person): Promise<Person> {
+    const session = this.driver.session();
+    const id = typeof person === "number" ? person : person.properties.id;
+
+    try {
+      const result = await session.run(
+        "MATCH (p:Person)-[:MOTHER]->(m:Person) WHERE p.id = $id RETURN m",
+        { id: id }
+      );
+
+      const mother = result.records[0].get("m");
+
+      return mother;
+    } finally {
+      await session.close();
+    }
+  }
+
+  async findPartners(person: number | Person): Promise<Person[]> {
+    const session = this.driver.session();
+    const id = typeof person === "number" ? person : person.properties.id;
+
+    try {
+      const result = await session.run(
+        "MATCH (p:Person)-[:PARTNER]-(partner:Person) WHERE p.id = $id RETURN partner",
+        { id: id }
+      );
+
+      const partners = result.records.map((record) => record.get("partner"));
+
+      return partners;
+    } finally {
+      await session.close();
+    }
+  }
+
+  async findChildren(person: number | Person): Promise<Person[]> {
+    const session = this.driver.session();
+    const id = typeof person === "number" ? person : person.properties.id;
+
+    try {
+      const result = await session.run(
+        "MATCH (p:Person)<-[:FATHER|:MOTHER]-(child:Person) WHERE p.id = $id RETURN child",
+        { id: id }
+      );
+
+      const children = result.records.map((record) => record.get("child"));
+
+      return children;
+    }
+    finally {
+      await session.close();
+    }
+  }
+
+  async updatePerson(person: PersonProperties): Promise<void> {
+    const session = this.driver.session();
+
+    try {
+      await session.run(
+        "MATCH (p:Person) WHERE p.id = $id SET p = $person",
+        { id: person.id, person }
+      );
+    } finally {
+      await session.close();
+    }
+  }
+
+  async createPerson(person: PersonProperties): Promise<void> {
+    const session = this.driver.session();
+
+    try {
+      await session.run(
+        "CREATE (p:Person $person)",
+        { person }
+      );
+    } finally {
+      await session.close();
+    }
+  }
+
+  async setFather(person: Person, father: Person): Promise<void> {
+    const session = this.driver.session();
+
+    try {
+      await session.run(
+        "MATCH (p:Person)-[r:FATHER]->(f:Person) WHERE p.id = $personId DELETE r",
+        { personId: person.properties.id }
+      );
+      await session.run(
+        "MATCH (p:Person), (f:Person) WHERE p.id = $personId AND f.id = $fatherId CREATE (p)-[:FATHER]->(f)",
+        { personId: person.properties.id, fatherId: father.properties.id }
+      );
+    } finally {
+      await session.close();
+    }
+  }
+
+  async setMother(person: Person, mother: Person): Promise<void> {
```

Comment:
As this function is similar to setFather and addPartner, why not combine them into setRelationship(person1, person2, relationship)? That would enable adoption for example...

Reply:
I kept it simple and using the model used by the front-end lib. We can change this but I wanted some placeholder code fast to connect the db so we could start working fast